File: api/src/endpoint/user/get_current_user.rs
use serde::{Deserialize, Serialize};
use api_derive::EndpointRequest;

use crate::{endpoint::Endpoint, resource::user::UserResource};

pub struct GetCurrentUser;

impl Endpoint for GetCurrentUser {
    const PATH: &'static str = "/user";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetCurrentUserRequest;
    type Response = GetCurrentUserResponse;
}

#[derive(EndpointRequest, Debug, Serialize, Deserialize)]
pub struct GetCurrentUserRequest {}

pub type GetCurrentUserResponse = UserResource;

File: gitdot-api-ts/src/endpoint/user/get-current-user.ts
import { z } from "zod";

import { UserResource } from "../../resource";

export const GetCurrentUserRequest = z.object({});
export type GetCurrentUserRequest = z.infer<typeof GetCurrentUserRequest>;

export const GetCurrentUser = {
  path: "/user",
  method: "GET",
  request: GetCurrentUserRequest,
  response: UserResource,
} as const;
export type GetCurrentUser = typeof GetCurrentUser;
